### DISCUSSION TITLE
Extra space not allowed after "end seq", only after "start seq"

### DISCUSSION THREAD

**Discussion by lilblam** [LostRuins/koboldcpp#312] [General]
When in instruct mode, and I set up the "start seq" and "end seq" in settings, very often the prompt is something like this:
```
### Instruction: {prompt}

### Response: 
```

There is a trailing space in the start seq, and the end seq. The start seq trailing space is respected, but the end seq trailing space is being trimmed. It still shows up in the UI field, but is clearly absent when you look at the command prompt when you test a model. Here is an example output:

Input: {"n": 1, "max_context_length": 2048, "max_length": 512, "rep_pen": 1.1, "temperature": 0.4, "top_p": 0.1, "top_k": 1, "top_a": 1, "typical": 1, "tfs": 1, "rep_pen_range": 256, "rep_pen_slope": 0.7, "sampler_order": [6, 0, 1, 2, 3, 4, 5], "prompt": "\n###Human: Hello model how are you\n###Assistant:", "quiet": true, "stop_sequence": ["###Human:", "###Assistant:", "\n\n\n\n\n"]}

Could the trailing space after ###Assistant: (in the End Seq field) not be trimmed?

Thanks!

**Comment by LostRuins** [comment]
This has already been added, what version are you currently running? I cannot remember if it's in 1.34.2 or will only be in 1.35, but the current newest code supports whitespaces in the instruct start and end tags.

**Reply by lilblam** [reply]
just upgraded to 1.35. I can't tell for sure, but I'll just trust that it's there and working as it should. I actually am not sure if a white space is needed after ### Assistant: or what not, someone suggested yes, but oogabooga has has a tooltip telling you not to add it, and the default oogabooga instruct templates don't seem to add it. So I'll just stop worrying about it lol
